src/algorithms/graph/detect-cycle/detectUndirectedCycleUsingDisjointSet.js

```diff
@@ -6,26 +6,19 @@ import DisjointSet from '../../../data-structures/disjoint-set/DisjointSet';
  * @param {Graph} graph
  */
 export default function detectUndirectedCycleUsingDisjointSet(graph) {
-  // Create initial singleton disjoint sets for each graph vertex.
-  /** @param {GraphVertex} graphVertex */
-  const keyExtractor = graphVertex => graphVertex.getKey();
-  const disjointSet = new DisjointSet(keyExtractor);
-  graph.getAllVertices().forEach(graphVertex => disjointSet.makeSet(graphVertex));
+  const vertices = graph.getAllVertices();
+  const disjointSet = new DisjointSet(vertices.length);
```

Comment:
I think it would be better not to lock Disjoint Set size from the very beginning but make it available to grow instead. Thus one of the basic Disjoint Set operation as "MakeSet" becomes possible.

Reply:
This seems reasonable & is easy to implement in the current scheme.